internal/provider/kubernetes/controller.go

```diff
@@ -1465,6 +1466,60 @@ func (r *gatewayAPIReconciler) processParamsRef(ctx context.Context, gc *gwapiv1
 				validationErr = fmt.Errorf("invalid envoyproxy: %w", err)
 				continue
 			}
+
+			if ep.Spec.Telemetry != nil {
+				telemetry := ep.Spec.Telemetry
+
+				if telemetry.AccessLog != nil {
+					for _, setting := range telemetry.AccessLog.Settings {
+						for _, sink := range setting.Sinks {
+							if sink.OpenTelemetry == nil {
+								continue
+							}
+							for _, backendRef := range sink.OpenTelemetry.BackendRefs {
+								backendNamespace := gatewayapi.NamespaceDerefOrAlpha(backendRef.Namespace, ep.Namespace)
+								resourceMap.allAssociatedBackendRefs[gwapiv1.BackendObjectReference{
+									Group:     backendRef.BackendObjectReference.Group,
+									Kind:      backendRef.BackendObjectReference.Kind,
+									Namespace: gatewayapi.NamespacePtrV1Alpha2(backendNamespace),
+									Name:      backendRef.Name,
+								}] = struct{}{}
+							}
+						}
+					}
+				}
+
+				if telemetry.Metrics != nil {
+					for _, sink := range telemetry.Metrics.Sinks {
+						if sink.OpenTelemetry == nil {
+							continue
+						}
+						for _, backendRef := range sink.OpenTelemetry.BackendRefs {
+							backendNamespace := gatewayapi.NamespaceDerefOrAlpha(backendRef.Namespace, ep.Namespace)
+							resourceMap.allAssociatedBackendRefs[gwapiv1.BackendObjectReference{
+								Group:     backendRef.BackendObjectReference.Group,
+								Kind:      backendRef.BackendObjectReference.Kind,
+								Namespace: gatewayapi.NamespacePtrV1Alpha2(backendNamespace),
+								Name:      backendRef.Name,
+							}] = struct{}{}
+						}
+					}
+				}
+
+				if telemetry.Tracing != nil {
+					for _, backendRef := range telemetry.Tracing.Provider.BackendRefs {
```

Comment:
the logic of processing backendRefs for accesslog/metrics/tracing are quite same, is there anyway to abstract them?

Reply:
need revist those struct later.